cli/azd/pkg/ext/models.go

```diff
@@ -0,0 +1,186 @@
+type HookType string
+
+type ScriptConfig struct {
+	validated bool
+
+	// Internal name of the hook running for a given command
+	Name string `yaml:",omitempty"`
+	// The type of script hook (bash or powershell)
+	Type ScriptType `yaml:"type,omitempty"`
+	// The location of the script hook (file path or inline)
+	Location ScriptLocation `yaml:"location,omitempty"`
+	// When location is `path` a file path must be specified relative to the project or service
+	Path string `yaml:"path,omitempty"`
+	// When location is `inline` a script must be defined inline
+	Script string `yaml:"script,omitempty"`
+	// When set to true will not halt command execution even when a script error occurs.
+	ContinueOnError bool `yaml:"continueOnError,omitempty"`
+	// When set to true will bind the stdin, stdout & stderr to the running console
+	Interactive bool `yaml:"interactive,omitempty"`
+	// When running on windows use this override config
+	Windows *ScriptConfig `yaml:"windows,omitempty"`
+	// When running on linux/macos use this override config
+	Posix *ScriptConfig `yaml:"linux,omitempty"`
+}
+
+// Validates and normalizes the script configuration
+func (sc *ScriptConfig) validate() error {
+	if sc.validated {
+		return nil
+	}
+
+	if sc.Type == ScriptTypeUnknown && sc.Path == "" {
+		return ErrScriptTypeUnknown
+	}
+
+	if sc.Location == ScriptLocationInline && sc.Script == "" {
+		return ErrScriptRequired
+	}
+
+	if sc.Location == ScriptLocationPath && sc.Path == "" {
+		return ErrPathRequired
+	}
+
+	if sc.Location == ScriptLocationUnknown {
+		if sc.Path != "" {
+			sc.Location = ScriptLocationPath
+		} else if sc.Script != "" {
+			sc.Location = ScriptLocationInline
+		}
+	}
+
+	if sc.Location == ScriptLocationInline {
+		tempScript, err := createTempScript(sc)
+		if err != nil {
+			return err
+		}
+
+		sc.Path = tempScript
+	}
+
+	if sc.Type == ScriptTypeUnknown {
+		scriptType, err := inferScriptTypeFromFilePath(sc.Path)
+		if err != nil {
+			return err
+		}
+
+		sc.Type = scriptType
+	}
+
+	_, err := os.Stat(sc.Path)
+	if err != nil {
+		return fmt.Errorf("script at '%s' is invalid, %w", sc.Path, err)
+	}
+
+	sc.validated = true
+
+	return nil
+}
+
+func inferScriptTypeFromFilePath(path string) (ScriptType, error) {
+	fileExtension := filepath.Ext(path)
+	switch fileExtension {
+	case ".sh":
+		return ScriptTypeBash, nil
+	case ".ps1":
+		return ScriptTypePowershell, nil
+	default:
+		return "", fmt.Errorf(
+			"script with file extension '%s' is not valid. %w.",
+			fileExtension,
+			ErrUnsupportedScriptType,
+		)
+	}
+}
+
+func createTempScript(scriptConfig *ScriptConfig) (string, error) {
+	var ext string
+	scriptHeader := []string{}
+
+	switch scriptConfig.Type {
+	case ScriptTypeBash:
+		ext = "sh"
+		scriptHeader = []string{
+			"#!/bin/sh",
+		}
+	case ScriptTypePowershell:
+		ext = "ps1"
+	}
+
+	// Creates .azure/hooks directory if it doesn't already exist
+	// In the future any scripts with names like "predeploy.sh" or similar would
```

Comment:
My preference is that we remove the entire implicit hooks feature (and when we do that shift to writing these inline scripts into `$TEMP`) but if we are going to keep it, we should update this comment because you implemented this feature...